fdbserver/workloads/BitFlipping.actor.cpp

```diff
@@ -28,85 +28,202 @@
+		if (res.isError()) {
+			auto err = res.getError();
+			if (err.code() == error_code_client_invalid_operation) {
+				TraceEvent(SevError, "ChaosDisabled")
+				    .detail("OnEndpoint", worker.waitFailure.getEndpoint().addresses.address.toString());
+			} else {
+				TraceEvent(SevError, "BitFlippingFailed")
+				    .detail("OnEndpoint", worker.waitFailure.getEndpoint().addresses.address.toString())
+				    .error(err);
+			}
+		}
+	}
+
+	ACTOR void doBitFlips(WorkerInterface worker, double percentage, double startDelay = 0.0) {
+		state Future<Void> res;
+		wait(::delay(startDelay));
+		SetFailureInjection::FlipBitsCommand flipBits;
+		flipBits.percentBitFlips = percentage;
+		SetFailureInjection req;
+		req.flipBits = flipBits;
+		res = worker.clientInterface.setFailureInjection.getReply(req);
+		wait(ready(res));
+		checkBitFlipResult(res, worker);
+	}
+
+	ACTOR static Future<Void> getAllStorageWorkers(Database cx,
+	                                               BitFlippingWorkload* self,
+	                                               std::vector<WorkerInterface>* result) {
+		result->clear();
+		state std::vector<WorkerInterface> res = wait(getStorageWorkers(cx, self->dbInfo, false));
+		for (auto& worker : res) {
+			result->emplace_back(worker);
+		}
+		return Void();
+	}
+
+	ACTOR template <class W>
+	Future<Void> flipBitsClient(Database cx, BitFlippingWorkload* self) {
+		state double lastTime = now();
+		state double workloadEnd = now() + self->testDuration;
+		state std::vector<W> machines;
+		loop {
+			wait(poisson(&lastTime, 1));
+			wait(BitFlippingWorkload::getAllStorageWorkers(cx, self, &machines));
+			auto machine = deterministicRandom()->randomChoice(machines);
+
+			// If we have already chosen this worker, then just continue
+			if (find(self->chosenWorkers.begin(), self->chosenWorkers.end(), machine.address()) !=
+			    self->chosenWorkers.end())
+				continue;
+
+			// Keep track of chosen workers for verification purpose
+			self->chosenWorkers.emplace_back(machine.address());
+			self->doBitFlips(machine, self->percentBitFlips);
+		}
+	}
+
+	// Resend the chaos event to previosuly chosen workers, in case some workers got restarted and lost their chaos
+	// config
+	ACTOR static Future<Void> reSendChaos(BitFlippingWorkload* self) {
+		std::vector<WorkerDetails> workers =
+		    wait(self->dbInfo->get().clusterInterface.getWorkers.getReply(GetWorkersRequest{}));
+		std::map<NetworkAddress, WorkerInterface> workersMap;
+		for (auto worker : workers) {
+			workersMap[worker.interf.address()] = worker.interf;
+		}
+		for (auto& workerAddress : self->chosenWorkers) {
+			auto itr = workersMap.find(workerAddress);
+			if (itr != workersMap.end())
+				self->doBitFlips(itr->second, self->percentBitFlips);
+		}
+		return Void();
+	}
+	// For fetching chaosMetrics to ensure chaos events are happening
+	// This is borrowed code from Status.actor.cpp
```

Comment:
`fdbserver/Status.h` exists so just make a prototype of the original rather than copying a large function.

Reply:
Done